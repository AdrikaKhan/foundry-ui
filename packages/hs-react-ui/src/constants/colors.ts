export default {
  background: '#fff',
  primary: '#e75b27',
  
  grayDark: '#252c35',
  grayMedium: '#51575d',
  grayLight: '#7c8186',
  grayXlight: '#d3d6d7',

  destructive: '#d00'
};
