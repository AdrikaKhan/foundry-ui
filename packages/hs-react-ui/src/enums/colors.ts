enum colors {
  background = '#fff',
  primary = '#e75b27',

  grayDark = '#252c35',
  grayMedium = '#51575d',
  grayLight = '#7c8186',
  grayXlight = '#d3d6d7',

  success = '#698F00',
  destructive = '#d00',

  blueCharcoal25 = 'rgba(211, 214, 215, 0.25)',
  blueCharcoal50 = 'rgba(211, 214, 215, 0.5)',
  blueCharcoal75 = 'rgba(211, 214, 215, 0.75)',
  blueCharcoal100 = 'rgba(211, 214, 215, 1)',
}

export default colors;
