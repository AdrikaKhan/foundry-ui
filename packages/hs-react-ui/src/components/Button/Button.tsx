import React, { ReactNode } from 'react';
import UnstyledIcon from '@mdi/react';
import { mdiLoading } from '@mdi/js';
import styled, { StyledComponentBase } from 'styled-components';
import { readableColor, darken } from 'polished';

import timings from '../../enums/timings';
import colors from '../../enums/colors';
import Progress from '../Progress/Progress';
import { Span, Button as ButtonElement } from '../../htmlElements';

export type ButtonContainerProps = {
  elevation: number;
  color: string;
  type: string;
};

export enum ButtonTypes {
  fill = 'fill',
  link = 'link',
  outline = 'outline',
}

export type ButtonProps = {
  StyledContainer?: string & StyledComponentBase<any, {}, ButtonContainerProps>;
  iconPrefix?: string | JSX.Element;
  iconSuffix?: string | JSX.Element;
  isLoading?: boolean;
  isProcessing?: boolean;
  children?: ReactNode;
  elevation?: number;
  type?: ButtonTypes;
  color?: string;
  onClick: (...args: any[]) => void;
  LoadingBar?: string & StyledComponentBase<any, {}>;
};

/**
 * Get the appropriate font color for the button based on the type of button
 * @param {string} type - The type of button
 * @param {string} color - The color prop passed into the button
 */
export const getFontColorFromType = (type: string, color: string) => {
  if (type === 'fill') {
    return readableColor(color, colors.background, colors.grayDark, true);
  }
  return color;
};

/**
 * Get the appropriate background color for the button based on the type of button
 * @param {string} type - The type of button
 * @param {string} color - The color prop passed into the button
 */
export const getBackgroundColorFromType = (type: string, color: string) => {
  switch (type) {
    case ButtonTypes.link:
    case ButtonTypes.outline:
      return colors.transparent;
    default:
      return color;
  }
};

export const ButtonContainer: string & StyledComponentBase<any, {}, ButtonContainerProps> = styled(
  ButtonElement,
)`
  ${({ elevation = 0, color, type }: ButtonContainerProps) => {
    const backgroundColor = getBackgroundColorFromType(type, color);
    const fontColor = getFontColorFromType(type, color);
    const shadowYOffset = elevation && elevation >= 1 ? (elevation - 1) * 0.5 + 0.1 : 0;
    const shadowBlur = elevation && elevation >= 1 ? (elevation - 1) * 0.5 + 0.1 : 0;
    const shadowOpacity = 0.5 - elevation * 0.075;

    return `
      display: inline-block;
      font-size: 1em;
      padding: .75em 1em;
      border-radius: 0.25em;
      transition: filter ${timings.slow};
      filter: drop-shadow(0em ${shadowYOffset}em ${shadowBlur}em rgba(0,0,0,${shadowOpacity}));
      outline: 0 none;
      border: ${type === ButtonTypes.outline ? `1px solid ${color || colors.grayDark}` : `0 none;`};
      cursor: pointer;
      background-color: ${backgroundColor};
      color: ${fontColor};
      &:hover {
        background-color: ${
          backgroundColor !== 'transparent' ? darken(0.05, backgroundColor) : 'rgba(0, 0, 0, 0.05)'
        };
      }
      &:active {
        background-color: ${
          backgroundColor !== 'transparent' ? darken(0.1, backgroundColor) : 'rgba(0, 0, 0, 0.1)'
        };
      }
    `;
  }}
`;

const StyledProgress = styled(Progress)`
  width: 5rem;
  height: 10px;
  margin-top: -5px;
  margin-bottom: -5px;
`;

const IconContainer = styled(Span)`
  margin-top: -8px;
  margin-bottom: -8px;
`;

const LeftIconContainer = styled(IconContainer)`
  margin-right: 1em;
`;

const RightIconContainer = styled(IconContainer)`
  margin-left: 1em;
`;

const Button = ({
  StyledContainer = ButtonContainer,
  iconPrefix,
  iconSuffix,
  isLoading,
  isProcessing,
  children,
  elevation = 0,
  type = ButtonTypes.fill,
  color = colors.grayDark,
  onClick,
  LoadingBar = StyledProgress,
}: ButtonProps) => {
  return isLoading ? (
    <StyledContainer
      data-test-id="hsui-button"
      onClick={onClick}
      elevation={elevation}
      color={color}
      type={type}
    >
      <LoadingBar />
    </StyledContainer>
  ) : (
    <StyledContainer
      data-test-id="hsui-button"
      onClick={onClick}
      elevation={elevation}
      color={color}
      type={type}
    >
      {!isProcessing &&
        iconPrefix &&
        (typeof iconPrefix === 'string' && iconPrefix !== '' ? (
          <LeftIconContainer>
            <UnstyledIcon path={iconPrefix} size="1rem" />
          </LeftIconContainer>
        ) : (
          <LeftIconContainer>{iconPrefix}</LeftIconContainer>
        ))}
      {isProcessing && (
        <LeftIconContainer>
          <UnstyledIcon path={mdiLoading} size="1rem" spin={1} />
        </LeftIconContainer>
      )}
      {children}

      {iconSuffix &&
        (typeof iconSuffix === 'string' ? (
          <RightIconContainer>
            <UnstyledIcon path={iconSuffix} size="1rem" />
          </RightIconContainer>
        ) : (
          <RightIconContainer>{iconSuffix}</RightIconContainer>
        ))}
    </StyledContainer>
  );
};

Button.Container = ButtonContainer;
Button.ButtonTypes = ButtonTypes;
Button.LoadingBar = StyledProgress;
export default Button;
