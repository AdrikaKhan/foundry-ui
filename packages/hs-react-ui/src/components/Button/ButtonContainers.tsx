import styled, { StyledComponentBase } from 'styled-components';
import { darken, readableColor } from 'polished';
import colors from '../../enums/colors';
import fonts from '../../enums/fonts';

export type ButtonContainerProps = {
  elevation?: number;
  color?: string;
  type: string;
};
export enum ButtonTypes {
  primary = 'primary',
  secondary = 'secondary',
  default = 'default',
  destructive = 'destructive',
  link = 'link',
  outline = 'outline',
}

const getFontColorFromType = (type?: string, color?: string) => {
  switch (type) {
    case ButtonTypes.primary:
    case ButtonTypes.secondary:
    case ButtonTypes.destructive:
      return color
        ? readableColor(color, colors.background, colors.grayDark, true)
        : colors.background;
    case ButtonTypes.link:
    case ButtonTypes.outline:
      return color || colors.grayDark;
    default:
      return color
        ? readableColor(color, colors.background, colors.grayDark, true)
        : colors.grayDark;
  }
};

const getBackgroundColorFromType = (type?: string, color?: string) => {
  switch (type) {
    case ButtonTypes.primary:
      return color || colors.primary;
    case ButtonTypes.secondary:
      return color || colors.grayMedium;
    case ButtonTypes.destructive:
      return color || colors.destructive;
    case ButtonTypes.link:
    case ButtonTypes.outline:
      return colors.transparent;
    default:
      return color || colors.grayXlight;
  }
};

// TODO: We need to use filter svg drop-shadow to get shadows for transparent (and all other) buttons
export const ButtonContainer: string &
  StyledComponentBase<any, {}, ButtonContainerProps> = styled.button`
  ${({ elevation = 0, color, type }: ButtonContainerProps) => {
    const backgroundColor = getBackgroundColorFromType(type, color);
    const fontColor = getFontColorFromType(type, color);
    const elevationYOffset = elevation && elevation >=1 ? (elevation - 1) * 0.5 + 0.1 : 0;
    const elevationBlur = elevation && elevation >= 1 ? (elevation / 16) * 0.1 + 0.3 : 0;
    return `
    display: inline-block;
    ${fonts.body}
    font-size: 1rem;
    line-height: 0;
    padding: 1.25em 1em;
    border-radius: 0.25em;
    filter: drop-shadow(0rem ${elevationYOffset}rem ${elevationBlur}rem rgba(0,0,0,${0.6 -
      elevation * 0.1}));
    outline: 0 none;
    border: ${type === ButtonTypes.outline ? `1px solid ${color || colors.grayDark}` : `0 none;`};
    cursor: pointer;
    background-color: ${backgroundColor};
    color: ${fontColor};
    &:hover {
      background-color: ${darken(0.05, backgroundColor)};
    }
    &:active {
      background-color: ${darken(0.1, backgroundColor)};
    }
  `;
  }}
`;

export default ButtonContainer;
