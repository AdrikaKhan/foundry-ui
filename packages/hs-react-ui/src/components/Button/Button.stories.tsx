import React from 'react';
import { action } from '@storybook/addon-actions';
import { select, text } from '@storybook/addon-knobs';

import Button from './Button';
import { ButtonTypes } from '../../enums/ButtonTypes';

export default {
  title: '',
  component: Button,
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/3r2G00brulOwr9j7F6JF59/Generic-UI-Style?node-id=83%3A2'
  }
};

export const Basic = () => (
  <Button
    buttonType={select('Button Type', ButtonTypes, ButtonTypes.primary)}
    onClick={action('button-click')}
  >
    {text('children', 'Default text')}
  </Button>
);

Basic.story = {
  name: 'Basic'
};