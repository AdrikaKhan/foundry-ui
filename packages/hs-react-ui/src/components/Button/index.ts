import Button from './Button'
import { ButtonContainer,
    PrimaryButtonContainer,
    SecondaryButtonContainer,
    DestructiveButtonContainer } from './ButtonContainers'

export {
  ButtonContainer,
  PrimaryButtonContainer,
  SecondaryButtonContainer,
  DestructiveButtonContainer
}
export default Button;