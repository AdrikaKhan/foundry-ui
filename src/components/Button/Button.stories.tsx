import * as React from 'react';
import styled from 'styled-components';

import { Story, Meta } from '@storybook/react';

import { mdiMessage, mdiSend } from '@mdi/js';

import Button, { ButtonProps } from './Button';

import variants from 'src/enums/variants';
import colors from 'src/enums/colors';
import FeedbackTypes from 'src/enums/feedbackTypes';

const iconOptions = {
  options: ['none', 'mdiMessage', 'mdiSend'],
  mapping: {
    none: '',
    mdiMessage,
    mdiSend,
  },
  control: {
    type: 'radio',
  },
};

const StyledIconContainer = styled.div`
  z-index: 1;
  margin-left: 1em;
  transition: transform 0.5s ease;
  &:hover {
    transform: rotate(180deg);
  }
`;

export const BasicButton: Story<ButtonProps> = args => <Button {...args} />;
BasicButton.args = {
  variant: variants.fill,
  color: colors.primaryDark,
  disabled: false,
  feedbackType: FeedbackTypes.ripple,
  isLoading: false,
  elevation: 1,
  isProcessing: false,
  iconPrefix: 'none',
  iconSuffix: 'none',
  children: 'Default text',
  StyledRightIconContainer: StyledIconContainer,
};
BasicButton.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/3r2G00brulOwr9j7F6JF59/Generic-UI-Style?node-id=83%3A2',
  },
};

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    iconPrefix: iconOptions,
    iconSuffix: iconOptions,
  },
} as Meta;
