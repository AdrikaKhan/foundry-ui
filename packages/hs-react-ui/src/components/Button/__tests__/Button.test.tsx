import React from 'react';
import { render, fireEvent, waitFor, configure } from '@testing-library/react';
import Icon from '@mdi/react';
import colors from '../../../enums/colors';
import Button, { ButtonTypes } from '../Button';

configure({ testIdAttribute: 'data-test-id' });

const testId = 'hsui-button';

describe('Button', () => {
  it('fires click handler when clicked', () => {
    const clickSpy = jest.fn();
    const { getByTestId } = render(<Button onClick={clickSpy}>Click me</Button>);

    fireEvent.click(getByTestId(testId));

    expect(clickSpy).toHaveBeenCalled();
  });

  it('shows loading text when provided', async () => {
    const { container, getByTestId } = render(<Button isLoading={true} onClick={() => {}} />);

    await waitFor(() => getByTestId(testId));

    expect(container).toMatchSnapshot();
  });

  it('shows ButtonContainer with nondefault props type and color', async () => {
    const { container, getByTestId } = render(<Button color={colors.black} elevation={2} type={ButtonTypes.outline} onClick={() => {}} />);

    await waitFor(() => getByTestId(testId));

    expect(container).toMatchSnapshot();
  });

  it('shows ButtonContainer ButtonType.link', async () => {
    const { container, getByTestId } = render(<Button type={ButtonTypes.link} onClick={() => {}} />);

    await waitFor(() => getByTestId(testId));

    expect(container).toMatchSnapshot();
  });

  it('shows ButtonContainer ButtonType.fill', async () => {
    const { container, getByTestId } = render(<Button type={ButtonTypes.fill} onClick={() => {}} />);

    await waitFor(() => getByTestId(testId));

    expect(container).toMatchSnapshot();
  });

  it('shows ButtonContainer ButtonType.outline', async () => {
    const { container, getByTestId } = render(<Button type={ButtonTypes.outline} color={null} onClick={() => {}} />);

    await waitFor(() => getByTestId(testId));

    expect(container).toMatchSnapshot();
  });

  it('shows LeftIconContainer when isProcessing', async () => {
    const { container, getByTestId } = render(<Button isProcessing onClick={() => {}} />);

    await waitFor(() => getByTestId(testId));

    expect(container).toMatchSnapshot();
  });

  it('shows icons with type string', async () => {
    const { container, getByTestId } = render(<Button iconSuffix='mdiComment' iconPrefix='mdiComment' onClick={() => {}} />);

    await waitFor(() => getByTestId(testId));

    expect(container).toMatchSnapshot();
  });

  it('shows icons', async () => {
    const { container, getByTestId } = render(<Button iconSuffix={<Icon path={'mdiComment'}/>} iconPrefix={<Icon path={'mdiComment'}/>} onClick={() => {}} />);

    await waitFor(() => getByTestId(testId));

    expect(container).toMatchSnapshot();
  });

  it('keeps the container the same when switching between isLoading and not isLoading', async () => {
    const onClick = jest.fn();
    const { getByTestId, rerender, asFragment } = render(
      <Button isLoading={true} onClick={onClick} />,
    );

    await waitFor(() => getByTestId(testId));
    const loadingFragment = asFragment();

    rerender(<Button onClick={onClick}>Submit</Button>);
    await waitFor(() => getByTestId(testId));
    const loadedFragment = asFragment();

    // TODO: Use toMatchDiffSnapshot() between the fragments once we can figure out
    //  how to make it use the jest-styled-components plugin
    expect(loadingFragment.firstChild).toMatchSnapshot();
    expect(loadedFragment.firstChild).toMatchSnapshot();
  });
});
