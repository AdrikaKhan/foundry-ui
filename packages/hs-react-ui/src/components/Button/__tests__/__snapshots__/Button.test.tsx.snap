// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button keeps the container the same when switching between isLoading and not isLoading 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #252c35;
  color: #fff;
}

.c0:hover {
  background-color: #1b2026;
}

.c0:active {
  background-color: #101317;
}

.c1 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  background: linear-gradient( 45deg, rgba(255,255,255,0.75), rgba(0,0,0,0.75), rgba(255,255,255,0.75), rgba(0,0,0,0.75), rgba(255,255,255,0.75) ) repeat;
  background-size: 400% 100%;
  width: 6rem;
  height: 16px;
  border-radius: 9999px;
  -webkit-animation: ia-dwnO 8s linear infinite;
  animation: ia-dwnO 8s linear infinite;
  line-height: 0;
  width: 5rem;
  height: 10px;
  margin-top: -5px;
  margin-bottom: -5px;
}

<button
  class="sc-AxhUy sc-AxgMl c0"
  color="#252c35"
  data-test-id="hsui-button"
  elevation="0"
  type="fill"
>
  <div
    class="sc-AxjAm sc-AxirZ sc-fzpjYC c1"
  />
</button>
`;

exports[`Button keeps the container the same when switching between isLoading and not isLoading 2`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #252c35;
  color: #fff;
}

.c0:hover {
  background-color: #1b2026;
}

.c0:active {
  background-color: #101317;
}

<button
  class="sc-AxhUy sc-AxgMl c0"
  color="#252c35"
  data-test-id="hsui-button"
  elevation="0"
  type="fill"
>
  Submit
</button>
`;

exports[`Button shows ButtonContainer ButtonType.fill 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #252c35;
  color: #fff;
}

.c0:hover {
  background-color: #1b2026;
}

.c0:active {
  background-color: #101317;
}

<div>
  <button
    class="sc-AxhUy sc-AxgMl c0"
    color="#252c35"
    data-test-id="hsui-button"
    elevation="0"
    type="fill"
  />
</div>
`;

exports[`Button shows ButtonContainer ButtonType.link 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: transparent;
  color: #252c35;
}

.c0:hover {
  background-color: rgba(0,0,0,0.05);
}

.c0:active {
  background-color: rgba(0,0,0,0.1);
}

<div>
  <button
    class="sc-AxhUy sc-AxgMl c0"
    color="#252c35"
    data-test-id="hsui-button"
    elevation="0"
    type="link"
  />
</div>
`;

exports[`Button shows ButtonContainer ButtonType.outline 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 1px solid #252c35;
  cursor: pointer;
  background-color: transparent;
  color: null;
}

.c0:hover {
  background-color: rgba(0,0,0,0.05);
}

.c0:active {
  background-color: rgba(0,0,0,0.1);
}

<div>
  <button
    class="sc-AxhUy sc-AxgMl c0"
    data-test-id="hsui-button"
    elevation="0"
    type="outline"
  />
</div>
`;

exports[`Button shows ButtonContainer with nondefault props type and color 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0.6em 0.6em rgba(0,0,0,0.35));
  filter: drop-shadow(0em 0.6em 0.6em rgba(0,0,0,0.35));
  outline: 0 none;
  border: 1px solid #000;
  cursor: pointer;
  background-color: transparent;
  color: #000;
}

.c0:hover {
  background-color: rgba(0,0,0,0.05);
}

.c0:active {
  background-color: rgba(0,0,0,0.1);
}

<div>
  <button
    class="sc-AxhUy sc-AxgMl c0"
    color="#000"
    data-test-id="hsui-button"
    elevation="2"
    type="outline"
  />
</div>
`;

exports[`Button shows LeftIconContainer when isProcessing 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #252c35;
  color: #fff;
}

.c0:hover {
  background-color: #1b2026;
}

.c0:active {
  background-color: #101317;
}

.c1 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  margin-top: -8px;
  margin-bottom: -8px;
  margin-right: 1em;
}

<div>
  <button
    class="sc-AxhUy sc-AxgMl c0"
    color="#252c35"
    data-test-id="hsui-button"
    elevation="0"
    type="fill"
  >
    <span
      class="sc-AxiKw sc-AxhCb sc-fznWqX c1"
    >
      <svg
        role="presentation"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
      >
        <style>
          @keyframes spin { to { transform: rotate(360deg) } }
        </style>
        <g
          style="animation: spin linear 1s infinite; transform-origin: center;"
        >
          <path
            d="M12,4V2A10,10 0 0,0 2,12H4A8,8 0 0,1 12,4Z"
            style="fill: currentColor;"
          />
          <rect
            fill="transparent"
            height="24"
            width="24"
          />
        </g>
      </svg>
    </span>
  </button>
</div>
`;

exports[`Button shows icons 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #252c35;
  color: #fff;
}

.c0:hover {
  background-color: #1b2026;
}

.c0:active {
  background-color: #101317;
}

.c1 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  margin-top: -8px;
  margin-bottom: -8px;
  margin-right: 1em;
}

.c2 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  margin-top: -8px;
  margin-bottom: -8px;
  margin-left: 1em;
}

<div>
  <button
    class="sc-AxhUy sc-AxgMl c0"
    color="#252c35"
    data-test-id="hsui-button"
    elevation="0"
    type="fill"
  >
    <span
      class="sc-AxiKw sc-AxhCb sc-fznWqX c1"
    >
      <svg
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="mdiComment"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <span
      class="sc-AxiKw sc-AxhCb sc-fznWqX c2"
    >
      <svg
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="mdiComment"
          style="fill: currentColor;"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Button shows icons with type string 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #252c35;
  color: #fff;
}

.c0:hover {
  background-color: #1b2026;
}

.c0:active {
  background-color: #101317;
}

.c1 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  margin-top: -8px;
  margin-bottom: -8px;
  margin-right: 1em;
}

.c2 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  margin-top: -8px;
  margin-bottom: -8px;
  margin-left: 1em;
}

<div>
  <button
    class="sc-AxhUy sc-AxgMl c0"
    color="#252c35"
    data-test-id="hsui-button"
    elevation="0"
    type="fill"
  >
    <span
      class="sc-AxiKw sc-AxhCb sc-fznWqX c1"
    >
      <svg
        role="presentation"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="mdiComment"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <span
      class="sc-AxiKw sc-AxhCb sc-fznWqX c2"
    >
      <svg
        role="presentation"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="mdiComment"
          style="fill: currentColor;"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Button shows loading text when provided 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  display: inline-block;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: filter .5s;
  transition: filter .5s;
  -webkit-filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  filter: drop-shadow(0em 0em 0em rgba(0,0,0,0.5));
  outline: 0 none;
  border: 0 none;
  cursor: pointer;
  background-color: #252c35;
  color: #fff;
}

.c0:hover {
  background-color: #1b2026;
}

.c0:active {
  background-color: #101317;
}

.c1 {
  box-sizing: border-box;
  font-family: Montserrat,Roboto,sans-serif;
  background: linear-gradient( 45deg, rgba(255,255,255,0.75), rgba(0,0,0,0.75), rgba(255,255,255,0.75), rgba(0,0,0,0.75), rgba(255,255,255,0.75) ) repeat;
  background-size: 400% 100%;
  width: 6rem;
  height: 16px;
  border-radius: 9999px;
  -webkit-animation: ia-dwnO 8s linear infinite;
  animation: ia-dwnO 8s linear infinite;
  line-height: 0;
  width: 5rem;
  height: 10px;
  margin-top: -5px;
  margin-bottom: -5px;
}

<div>
  <button
    class="sc-AxhUy sc-AxgMl c0"
    color="#252c35"
    data-test-id="hsui-button"
    elevation="0"
    type="fill"
  >
    <div
      class="sc-AxjAm sc-AxirZ sc-fzpjYC c1"
    />
  </button>
</div>
`;
